src/components/TrackingProvider/tests/TrackingProvider.test.js

```diff
@@ -88,14 +88,14 @@ const CONTEXT_DATA = {
 };
 
 const MERGED_DATA = {
-    eventFields: {...CONTEXT_DATA.eventFields, ...PROP_DATA.eventFields},
+    eventPayload: {...CONTEXT_DATA.eventPayload, ...PROP_DATA.eventPayload},
```

Comment:
Can you add a test or two in this file verifying that `fields` and `eventFields` still gets used if `payload` and `eventPayload` are not specified and also that if both are specified then `payload` / `eventPayload` take precedence.